### DISCUSSION TITLE
Is there a way to get Pulse working with NIO.

### DISCUSSION THREAD

**Discussion by j-j-m** [kean/Pulse#291] [Q&A]
My application uses https://github.com/apple/swift-openapi-generator
to generate its API. Is there a way to integrate this with pulse given it is backed by NIO?

**Comment by j-j-m** [comment]
for those who land on this in the future https://github.com/apple/swift-openapi-urlsession

**Reply by kean** [reply]
That sould work as Pulse is primarily focused on `URLSession` APIs. You can learn a bit more about it in https://kean-docs.github.io/pulse/documentation/pulse/networklogging-article.
